### ISSUE TITLE
[BUG] x-develop : watch : sync processes can stack up

### ISSUE THREAD

**Issue by phpguru** [docker/compose#10605]
### Description

Occasionally I will start new terminal panes, and then also start one in the IDE as well, then forget where I am syncing the files. It's possible for alpha watch to get stuck in a state where multiple instances of the watch sync can be running. 

As I mentioned [here](https://github.com/docker/compose/issues/10358#issuecomment-1559633453) this feature is a godsend. Massive performance increase, and I don't have to install NFS or fiddle with fswatch, rsync or mutagen, or install ftp service into container.  

### Steps To Reproduce

1. Create any collection of services in your docker-compose.yml
2. Run `docker compose alpha watch` in multiple terminals
3. Run `ps aux | grep docker | grep alpha | grep watch` to see multiple processes

I usually `sudo kill -9 <pid>` on each of them to completely remove it, then start fresh

When you have multiple running, sometimes the changed file can get copied multiple times also.

### Compose Version

```Text
v2.17.3
```

### Docker Environment

```Text
Client:
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.9.1)
  compose: Docker Compose (Docker Inc., v2.17.3)
  dev: Docker Dev Environments (Docker Inc., v0.1.0)
  extension: Manages Docker extensions (Docker Inc., v0.2.19)
  init: Creates Docker-related starter files for your project (Docker Inc., v0.1.0-beta.4)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.26.0)
  scout: Command line tool for Docker Scout (Docker Inc., v0.10.0)
WARNING: Plugin "/Users/phpguru/.docker/cli-plugins/docker-compose-2.14.0" is not valid: plugin candidate "compose-2.14.0" did not match "^[a-z][a-z0-9]*$"
WARNING: Plugin "/Users/phpguru/.docker/cli-plugins/docker-compose-darwin-aarch64" is not valid: plugin candidate "compose-darwin-aarch64" did not match "^[a-z][a-z0-9]*$"

Server:
 Containers: 3
  Running: 3
  Paused: 0
  Stopped: 0
 Images: 58
 Server Version: 23.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc version: v1.1.5-0-gf19387a
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.49-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 6
 Total Memory: 15.61GiB
 Name: docker-desktop
 ID: 7f79dbcb-7b2f-4488-8003-0f0dae9a7582
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Anything else?

```
~/Projects/spoton-environment (DBT-341-permissions) $ ps aux | grep docker | grep watch | grep alpha
phpguru     82840   0.0  0.1 409394224  26336 s000  S+    5:59AM   0:00.52 /Users/phpguru/.docker/cli-plugins/docker-compose compose --profile teamwork alpha watch
phpguru     82838   0.0  0.0 409240736  15152 s000  S+    5:59AM   0:00.32 /Applications/Docker.app/Contents/Resources/bin/com.docker.cli compose --profile teamwork alpha watch
phpguru     82837   0.0  0.1 409279808  26224 s000  S+    5:59AM   0:00.78 docker compose --profile teamwork alpha watch
phpguru     85996   0.0  0.1 409385120  35968 s001  S+    8:20AM   0:00.04 /Users/phpguru/.docker/cli-plugins/docker-compose compose --profile teamwork alpha watch
phpguru     85994   0.0  0.1 409261696  27232 s001  S+    8:20AM   0:00.05 docker compose --profile teamwork alpha watch
```

**Comment by ndeloof** [comment]
each `docker compose alpha watch` run register a filesystem watcher, I don't expect there's nothing we can do to prevent concurrent runs of the same command to trigger multiple `docker cp` as Docker Compose is stateless by design

**Comment by phpguru** [comment]
To be honest, it may not be a bug. I just associated multiple instances with multiple copies of the same file getting shipped into the container but having not made any additional changes - like 3 or 4 copies at a time. I suppose that may be a better bug report than this one.

**Comment by milas** [comment]
See also:
* https://github.com/docker/compose/issues/10754

**Comment by milas** [comment]
This will be fixed in the next Compose release. The `alpha watch` command now creates a project-specific PID file and will refuse to run if there's another _active_ process for the same project.

**Comment by milas** [comment]
This is now available in Compose v2.21.0. If you're using Docker Desktop, upgrade to v4.23.0 (you might need to manually check for updates in settings, it's still rolling out for automatic upgrade). Thanks for the suggestion!
